apc_mmap.c

```diff
@@ -51,7 +55,99 @@
 # define MAP_ANON MAP_ANONYMOUS
 #endif
 
-void *apc_mmap(char *file_mask, size_t size)
+#if defined(__linux__)
+static int apc_mmap_hugetlb_flags(char *mmap_hugetlb_mode)
+{
+	int flags;
+
+	if (!mmap_hugetlb_mode
+		|| (mmap_hugetlb_mode && !strlen(mmap_hugetlb_mode))
+		|| !strcasecmp(mmap_hugetlb_mode, "none")) {
+		return 0;
+	}
+
+# ifndef MAP_HUGETLB
+	flags = 0;
+	apc_warning("MAP_HUGETLB is not defined on this system");
+# else
+	flags = MAP_HUGETLB;
+	if (!strcasecmp(mmap_hugetlb_mode, "default")) {
+		return flags;
+	}
+#  ifdef MAP_HUGE_16KB
+	if (!strcasecmp(mmap_hugetlb_mode, "16kb")) {
```

Comment:
Instead of listing all of these individually, can we make use of MAP_HUGE_SHIFT instead?